Add Programming for Kids

HollyAdele opened the pull request:
<!-- Please fill in the **bold** fields and tick all applicable boxes by placing an "x" inside "[ ]". -->

**https://github.com/HollyAdele/awesome-programming-for-kids**

**Resources to teach programming to adults are very different from ones useful for teaching the basics to children.  Awesome lists of resources for adults are already on the Awesome list; here is a list of the best I've found for children.**

# ALL THE BELOW CHECKBOXES SHOULD BE DONE AND CHECKED AND THE ABOVE PLACEHOLDERS SHOULD BE FILLED IN BEFORE SUBMITTING A PULL REQUEST!

- [x] I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- [x] This pull request has a descriptive title. *(For example: `Add Node.js`)*
- The list I added
	- [x] **has been around for at least 20 days,**
	- [x] is a non-generated Markdown file in a GitHub repo,
	- [x] is not a duplicate,
	- [x] only has awesome items with descriptions *(Awesome lists are curations of the best, not of everything)*,
	- [x] contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
	- [x] has a Table of Contents section named `Contents` as the first section,
	- [x] has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) *(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)*,
	- [x] has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
	- [x] has been checked for inconsistent formatting, spelling, and grammar *(This includes starting each link description with an uppercase character and ending it with a period `.`, and consistent naming)*,
	- [x] has been added at the bottom of the appropriate category.

sindresorhus commented:
Awesome! Thanks for making it. I've been hoping for someone to do this.

Are there any places on the web kids can discuss programming with each other? Like a forum or chat. If so, that would be useful to include. Maybe even a place kids could get mentoring from experienced programmers.

HollyAdele commented:
There is, apparently!  

I added DIY.org, which has a comment feature (and way more programming content than I thought.)  I also changed the description on Khan Academy to include the Q&A feature.
